marshal.go

```diff
@@ -113,10 +113,6 @@ type Unmarshaler interface {
 //
 // The marshal/unmarshal error provides a list of supported types when an unsupported type is attempted.
 func Marshal(info TypeInfo, value interface{}) ([]byte, error) {
-	if info.Version() < protoVersion1 {
-		panic("protocol version not set")
-	}
```

Comment:
we probably want to keep this, IIUC it's checking if the field is set or left as empty